### PR TITLE
Move Settings Models to the Settings SDK

### PR DESCRIPTION
**Pull request by cbgbt**
*Description of changes:*
This moves the following packages from https://github.com/bottlerocket-os/bottlerocket to this repository:
* All current settings extension packages
* Any settings models not yet ported to settings extensions
* `modeled-types`, useful Rust types that can be composed into new settings models
* `model-derive` macro for auto-implementing useful string traits in settings models
* `scalar` and `scalar-derive` macro to assist modeling scalar values in settings models

This change was constructed by copying content and git history from the Bottlerocket repository directly, and then making discrete changes to those contents as a series of commits -- the commit history is very useful in reviewing this change.

From a user perspective, the `bottlerocket-settings-models` package is meant to expose the public API of our existing models, as well as tools used to build new models.

*Pending*
* [x] Build Bottlerocket repository with dependencies on this code
* [x] Fix comments/documentation in the contents of these files

---
This is the script that was used to import git history from bottlerocket-os/bottlerocket:

```
#!/usr/bin/env bash

set -e

cd /tmp
DIR="${1:=/tmp/sdk-models}"
BR=$DIR/bottlerocket
SDK=$DIR/bottlerocket-settings-sdk

BOTTLEROCKET_BRANCH=deleteme-models-export
SDK_IMPORT_BRANCH=deleteme-models-import
SDK_FINAL_BRANCH=vend-plugins-with-history

rm -rf $DIR

git clone git@github.com:bottlerocket-os/bottlerocket.git $BR

git clone git@github.com:bottlerocket-os/bottlerocket-settings-sdk.git $SDK
cd $SDK
git remote remove origin
git remote add origin git@github.com:cbgbt/bottlerocket-settings-sdk.git
git remote add BECAREFUL_UPSTREAM git@github.com:bottlerocket-os/twoliter.git

cd $BR
git checkout -b $BOTTLEROCKET_BRANCH
git fast-export $BOTTLEROCKET_BRANCH -- \
  sources/models \
  sources/settings-extensions > $DIR/export.fi

cd $SDK
git switch --orphan $BOTTLEROCKET_BRANCH
git fast-import < $DIR/export.fi
git checkout develop
git checkout -b $SDK_IMPORT_BRANCH
git merge --signoff --allow-unrelated-histories -m'chore: merge models history' $BOTTLEROCKET_BRANCH

mkdir -p bottlerocket-settings-models

mv sources/models/* bottlerocket-settings-models/
mv sources/settings-extensions/ bottlerocket-settings-models/

git add --all
git rm sources/models/.gitignore
git commit -m'chore: move models into desired directories'
rm -rf sources/

```

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: bottlerocket-settings-models/settings-extensions/ecs/Cargo.toml [hunk 1]
```diff
@@ -10,10 +10,6 @@ publish = false
 env_logger = "0.10"
```
Comment (sam-berning) [inline bottlerocket-settings-models/settings-extensions/ecs/Cargo.toml]: nit: `env_logger` not sorted correctly here

File: bottlerocket-settings-models/settings-extensions/motd/Cargo.toml [hunk 2]
```diff
@@ -7,11 +7,7 @@ edition = "2021"
 publish = false
 
 [dependencies]
```
Comment (sam-berning) [inline bottlerocket-settings-models/settings-extensions/motd/Cargo.toml:9]: `motd` setting extension still needs `env_logger`, worth doing in this commit so we don't forget?

**Review by sam-berning — APPROVED** [review]
This looks really good! Just a couple nits on the settings extensions.

**Comment by cbgbt** [comment]
* Force pushed to add some docstrings missing on new modules
* Added commits for fixes pointed out by @sam-berning

File: Cargo.toml [hunk 3]
```diff
@@ -2,9 +2,44 @@
...
+
+    # Public API for Bottlerocket settings models
+    "bottlerocket-settings-models/settings-models",
+
+    # Useful libraries for constructing settings models
+    "bottlerocket-settings-models/model-derive",
+    "bottlerocket-settings-models/modeled-types",
+    "bottlerocket-settings-models/scalar",
+    "bottlerocket-settings-models/scalar-derive",
+    "bottlerocket-settings-models/string_impls_for",
```
Comment (bcressey) [inline Cargo.toml]: nit: can we make this `string-impls-for` as part of this change?

**Review by bcressey — APPROVED** [review]

**Comment by cbgbt** [comment]
I'm re-structuring this PR To bring the git history from the bottlerocket-os/bottlerocket with the contents that are copied.

**Comment by cbgbt** [comment]
* Adds bottlerocket-os/bottlerocket history for `sources/models` and `sources/settings-extensions`
* Renames `string_impls_for` to `string_impls_for` -- @bcressey

**Review by bcressey — APPROVED** [review]

**Review by webern — APPROVED** [review]

**Pull request merged**